### PR TITLE
Raghav/routing

### PR DESCRIPTION
**Pull request by rakatyal**

### CODE REVIEW

File: npm-search/assets/css/base.css [hunk 1]
```diff
@@ -5,6 +5,11 @@ body {
   margin: 0; 
 } 
 
+h1 {
+  -webkit-margin-before: 0.3em;
```
Comment (nikhilkh) [inline npm-search/assets/css/base.css]: Why is this change needed?

Comment (rakatyal) [inline npm-search/assets/css/base.css]: I added the <h1> tags on the page for SEO purposes. But the default margins above and below were huge and our page title didn't look good so I reduced it.

File: npm-search/assets/js/app.js [hunk 2]
```diff
@@ -168,7 +168,15 @@ var Plugin = React.createClass({
             </li>
         )
     }
-})
+});
+
+var timer=0;
```
Comment (nikhilkh) [inline npm-search/assets/js/app.js]: Why do we need a global variable here?

File: npm-search/assets/js/app.js [hunk 3]
```diff
@@ -168,7 +168,15 @@ var Plugin = React.createClass({
...
+window.onpopstate = function(e){
```
Comment (nikhilkh) [inline npm-search/assets/js/app.js]: should `window.addEventHandler` be used?

File: npm-search/assets/js/app.js (hunk 2 above)
Comment (rakatyal) [inline npm-search/assets/js/app.js]: We could have added it as a state variable to the plugin list component, but that would unnecessarily complicate stuff.

File: npm-search/assets/js/app.js (hunk 3 above)
Comment (rakatyal) [inline npm-search/assets/js/app.js]: You mean bind this function to the onpopstate event using EventListener?

Comment (nikhilkh) [inline npm-search/assets/js/app.js]: yes, I meant `addEventListener` - that's just the best practice

File: npm-search/assets/js/app.js [hunk 4]
```diff
@@ -280,12 +299,23 @@ var App = React.createClass({
...
+    loadFilterText : function(filterText) {
+        this.setState(function(previousState, currentProps) {
+            return {
+                filterText: filterText,
+                plugins: previousState.plugins
+            };
+        });
+    },
+    getURLParameter : function(name) {
+        return decodeURIComponent((new RegExp('[?|&]' + name + '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20'))||null;
```
Comment (nikhilkh) [inline npm-search/assets/js/app.js]: consider adding a new line

File: npm-search/assets/js/app.js (hunk 2 above)
Comment (nikhilkh) [inline npm-search/assets/js/app.js]: I see - Can timer be a member of `PluginList` class?  Also, you might want to initialize to null instead of 0.

File: npm-search/assets/css/base.css (hunk 1 above)
Comment (nikhilkh) [inline npm-search/assets/css/base.css]: We should consider removing the wrapping altogether. Also, this change is webkit specific and likely does not work on IE.

File: npm-search/assets/js/app.js (hunk 3 above)
Comment (rakatyal) [inline npm-search/assets/js/app.js]: okay will update

File: npm-search/assets/css/base.css (hunk 1 above)
Comment (rakatyal) [inline npm-search/assets/css/base.css]: yeah removing it and bringing the text to a single line centrally aligned with the image

File: npm-search/assets/js/app.js (hunk 2 above)
Comment (rakatyal) [inline npm-search/assets/js/app.js]: MAking it a member of PluginList class will make it a state variable meaning we will have to update it's state every time we render PluginList which is not really required.
